adapters/beintoo/beintoo.go

```diff
@@ -0,0 +1,255 @@
+	"net/url"
+	"strconv"
+	"time"
+
+	"github.com/mxmCherry/openrtb"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/errortypes"
+	"github.com/prebid/prebid-server/openrtb_ext"
+)
+
+type BeintooAdapter struct {
+	endpoint string
+	testing  bool
+}
+
+func buildEndpoint(endpoint string, testing bool, timeout int64) string {
+
+	if timeout == 0 {
+		timeout = 1000
+	}
+
+	uriObj, _ := url.Parse(endpoint)
+	parameters := url.Values{}
+
+	if testing {
+
+		parameters.Add("t", "1000")
+		parameters.Add("ts", "2060541160")
+
+	} else {
+
+		parameters.Add("t", strconv.FormatInt(timeout, 10))
+		parameters.Add("ts", strconv.FormatInt(time.Now().Unix(), 10))
+
+	}
+
+	uriObj.RawQuery = parameters.Encode()
+	return uriObj.String()
+}
+
+func (a *BeintooAdapter) MakeRequests(request *openrtb.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	var errs []error
+
+	if len(request.Imp) == 0 {
+		return nil, []error{&errortypes.BadInput{
+			Message: fmt.Sprintf("No Imps in Bid Request"),
+		}}
+	}
+
+	if errs := preprocess(request); errs != nil && len(errs) > 0 {
+		return nil, append(errs, &errortypes.BadInput{
+			Message: fmt.Sprintf("Error in preprocess of Imp, err: %s", errs),
+		})
+	}
+
+	data, err := json.Marshal(request)
+	if err != nil {
+		return nil, []error{&errortypes.BadInput{
+			Message: fmt.Sprintf("Error in packaging request to JSON"),
+		}}
+	}
+
+	headers := http.Header{}
+	headers.Add("Content-Type", "application/json;charset=utf-8")
+	headers.Add("Accept", "application/json")
+
+	if request.Device != nil {
+		addHeaderIfNonEmpty(headers, "User-Agent", request.Device.UA)
+		addHeaderIfNonEmpty(headers, "X-Forwarded-For", request.Device.IP)
+		addHeaderIfNonEmpty(headers, "Accept-Language", request.Device.Language)
+		if request.Device.DNT != nil {
+			addHeaderIfNonEmpty(headers, "DNT", strconv.Itoa(int(*request.Device.DNT)))
+		}
+	}
+	if request.Site != nil {
+		addHeaderIfNonEmpty(headers, "Referer", request.Site.Page)
+	}
+
+	url := buildEndpoint(a.endpoint, a.testing, request.TMax)
+
+	return []*adapters.RequestData{{
+		Method:  "POST",
+		Uri:     url,
+		Body:    data,
+		Headers: headers,
+	}}, errs
+}
+
+func unpackImpExt(imp *openrtb.Imp) (*openrtb_ext.ExtImpBeintoo, error) {
+	var bidderExt adapters.ExtImpBidder
+	if err := json.Unmarshal(imp.Ext, &bidderExt); err != nil {
+		return nil, &errortypes.BadInput{
+			Message: err.Error(),
+		}
+	}
+
+	var BeintooExt openrtb_ext.ExtImpBeintoo
+	if err := json.Unmarshal(bidderExt.Bidder, &BeintooExt); err != nil {
+		return nil, &errortypes.BadInput{
+			Message: fmt.Sprintf("ignoring imp id=%s, invalid ImpExt", imp.ID),
+		}
+	}
+
+	tagIDValidation, err := strconv.ParseInt(BeintooExt.TagID, 10, 64)
+	if err != nil || tagIDValidation == 0 {
+		return nil, &errortypes.BadInput{
+			Message: fmt.Sprintf("ignoring imp id=%s, invalid tagid must be a String of numbers", imp.ID),
+		}
+	}
+
+	
+
+	return &BeintooExt, nil
```

Comment:
`gofmt` is still complaining about lines 117 and 118, please remove them and run `gofmt adapters/beintoo/beintoo.go` before commiting in order to make sure the tests pass.
![image](https://user-images.githubusercontent.com/15863840/81416090-3fddd780-9117-11ea-9241-9cfaa2982b5e.png)

Reply:
I tried to do the test and it doesn't show any other errors